org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/DocumentProviderTest.xtend

```diff
@@ -18,12 +21,124 @@ class DocumentProviderTest extends AbstractGitTest {
 
 	@Test
 	def void createCommitsNewFile() {
+		// given
+		val numberOfCommitsBefore = git.log.call.size
+
 		// when
 		documentProvider.create('a.txt', 'test')
 
 		// then
-		val commits = git.log.call
-		commits.assertSize(2)
+		val numberOfCommitsAfter = git.log.call.size
+		numberOfCommitsAfter.assertEquals(numberOfCommitsBefore + 1)
+	}
+
+	@Test
+	def void createPushesChanges() {
+		// given
+		val numberOfCommitsBefore = remoteGit.log.call.size
+		val newFileName = "theNewFile.txt"
+
+		// when
+		documentProvider.create(newFileName, 'test')
+
+		// then
+		remoteGit.assertSingleCommit(numberOfCommitsBefore, ADD, newFileName)
 	}
 
+	@Test
+	def void createOrUpdateNonExistingFileAddsAndCommits() {
+		// given
+		val nonExistingFile = "aFileThatHasNotYetBeenCreated.txt"
+		val numberOfCommitsBefore = git.log.call.size
+
+		// when
+		documentProvider.createOrUpdate(nonExistingFile, "Contents of new file")
+
+		// then
+		git.assertSingleCommit(numberOfCommitsBefore, ADD, nonExistingFile)
+	}
+
+	@Test
+	def void createOrUpdatePreExistingFileCommitsModifications() {
+		// given
+		val preExistingFile = preExistingFileInRepo
+		val numberOfCommitsBefore = git.log.call.size
+
+		// when
+		documentProvider.createOrUpdate(preExistingFile, "New contents of pre-existing file")
+
+		// then
+		git.assertSingleCommit(numberOfCommitsBefore, MODIFY, preExistingFile)
+	}
+	
+	@Test
+	def void saveCommitsChanges() {
+		// given
+		val existingFileName = preExistingFileInRepo
+		val numberOfCommitsBefore = git.log.call.size
+
+		// when
+		documentProvider.save(existingFileName, "Contents of pre-existing file")
+
+		// then
+		git.assertSingleCommit(numberOfCommitsBefore, MODIFY, existingFileName)
+	}
+
+	@Test
+	def void savePushesChanges() {
+		// given
+		val existingFileName = preExistingFileInRepo
+		val numberOfCommitsBefore = remoteGit.log.call.size
+
+		// when
+		documentProvider.save(existingFileName, "Contents of pre-existing file")
```

Comment:
should be "new contents for existing file" (as you did in createOrUpdatePreExistingFileCommitsModifications)

Reply:
Right, applies for both `saveCommitsChanges()` and `savePushesChanges()`.